pkg/stream/edgedexecconnection.go

```diff
@@ -39,49 +41,82 @@ func (e *EdgedExecConnection) CacheTunnelMessage(msg *Message) {
 	e.ReadChan <- msg
```

Comment:
from your fix, I do not know how to prevent write to the closed readChan

Reply:
can we add a flag to indicate the channel is closed then just return

Reply:
This would be a way.
This is the way I took it: M receivers, N senders, any one of them say "let's end the game" by notifying a moderator to close an additional signal channel.
FIY: https://go101.org/article/channel-closing.html